src/Renderer/LayeredMaterial.js

```diff
@@ -99,11 +82,8 @@ const LayeredMaterial = function LayeredMaterial(options) {
         vsOptions = '\n#define DATA_TEXTURE_ELEVATION\n';
     }
 
-    // see GLOBE FS
-    this.fragmentShaderHeader += getColorAtIdUv(nbSamplers);
-
-    this.fragmentShader = this.fragmentShaderHeader + TileFS;
-    this.vertexShader = PrecisionQualifier + vsOptions + TileVS;
+    this.fragmentShader = ShaderUtils.unrollLoops(this.fragmentShaderHeader + TileFS, { TEX_UNITS: nbSamplers });
```

Comment:
Using `this.defines`, this simplifies to `this.fragmentShader = ShaderUtils.unrollLoops(TileFS, this.defines);`